e2e/suites/graphs.test.ts

```diff
@@ -0,0 +1,127 @@
+    await proxyMock();
+    await adminLogin();
+  });
+
+  it('should create a basic page', async () => {
+    await expect(page).toClick('a', { text: 'Account settings' });
+    await expect(page).toClick('a', { text: 'Pages' });
+    await expect(page).toClick('a', { text: 'Add page' });
+    await expect(page).toFill('input[name="page.data.title"]', 'Page data viz');
+    await expect(page).toFill(localSelectors.pageContentsInput, '<Dataset />');
+    await expect(page).toMatchElement('button', { text: 'Save' });
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should insert Bar chart graph in created page', async () => {
+    const pageContentsInput = await page.$eval(localSelectors.pageContentsInput, el => (<HTMLInputElement>el).value);
+    expect(pageContentsInput).toContain('<Dataset />');
+
+    await page.type(localSelectors.pageContentsInput, graphs.barChart);
+    await expect(page).toMatchElement('button', { text: 'Save' });
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should display Bar chart graph in page with no more than a 3% difference', async () => {
+    const createdPageLink = await page.$eval(localSelectors.createdPageLink, el => (<HTMLLinkElement>el).href);
+    await page.goto(createdPageLink);
+    await page.waitFor(2000); // wait for the chart visualization animations to end
+
+    const chartContainer = await page.$('.recharts-responsive-container');
+    // @ts-ignore
+    const chartScreenshot = await chartContainer.screenshot();
+
+    expect(chartScreenshot).toMatchImageSnapshot({
+      failureThreshold: 0.03,
+      failureThresholdType: 'percent',
+      allowSizeMismatch: true,
+    });
+  });
+
+  it('should navigate back to the edit page and insert a Pie chart graph', async () => {
+    await page.goBack();
+    await page.click(localSelectors.pageContentsInput, {clickCount: 3});
+    await page.keyboard.press('Backspace');
+    await expect(page).toFill(localSelectors.pageContentsInput, '<Dataset />');
+    await page.type(localSelectors.pageContentsInput, graphs.pieChart);
+    await expect(page).toMatchElement('button', { text: 'Save' });
+    await expect(page).toClick('button', { text: 'Save' });
+    await expect(page).toMatch('Saved successfully.');
+    await expect(page).toMatch('(view page)');
+  });
+
+  it('should display Pie chart graph in page with no more than a 3% difference', async () => {
+    const createdPageLink = await page.$eval(localSelectors.createdPageLink, el => (<HTMLLinkElement>el).href);
+    await page.goto(createdPageLink);
+    await page.waitFor(2000);
+
+    const chartContainer = await page.$('.recharts-responsive-container');
+    // @ts-ignore
+    const chartScreenshot = await chartContainer.screenshot();
```

Comment:
@elric-wamugu i think we should no use ts-ignore unless we abosolutely need it, this case can be solved with a condition or using the ensure utility. same for the rest of @ts-ignore on the same case.

```
const chartContainer = ensure<ElementHandle>(await page.$('.ListChart '))
```